src/algorithms/hnsw.rs

```diff
@@ -509,6 +531,61 @@ pub fn local_search(
     results
 }
 
+fn local_search_vbase<'mmap, 'vector>(
+    mmap: &'mmap HnswMmap,
+    range: usize,
+    s: u32,
+    vector: &'vector [Scalar],
+) -> HnswIndexIter<'mmap, 'vector> {
+    assert!(range > 0);
+    let mut visited_guard = mmap.visited.fetch();
+    let mut visited = visited_guard.fetch();
+    let mut candidates = BinaryHeap::<Reverse<(Scalar, u32)>>::new();
+    let mut results = Heap::new(range);
+    let mut heap = BinaryHeap::<Reverse<HeapElement>>::new();
+    visited.mark(s);
+    let s_dis = mmap.quantization.distance(mmap.d, vector, s);
+    candidates.push(Reverse((s_dis, s)));
+    results.push(HeapElement {
```

Comment:
Push to `heap`.

A vec to be sorted is better here.

Reply:
Push `s` to heap.

`heap` need to be sorted.

Compare head of `heap` and `result` to return the next result .

Reply:
Fixed in new commit. Could you review again?